drivers/hdc1000/hdc1000.c

```diff
@@ -111,19 +119,42 @@ int hdc1000_get_results(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
 
     if (status == HDC1000_OK) {
         /* if all ok, we convert the values to their physical representation */
+        uint16_t traw = ((uint16_t)buf[0] << 8) | buf[1];
+        temp_cached = (int16_t)((((int32_t)traw * 16500) >> 16) - 4000);
+
+        uint16_t hraw = ((uint16_t)buf[2] << 8) | buf[3];
+        hum_cached  = (int16_t)(((int32_t)hraw * 10000) >> 16);
+
         if (temp) {
-            uint16_t traw = ((uint16_t)buf[0] << 8) | buf[1];
-            *temp = (int16_t)((((int32_t)traw * 16500) >> 16) - 4000);
+            *temp = temp_cached;
         }
         if (hum) {
-            uint16_t hraw = ((uint16_t)buf[2] << 8) | buf[3];
-            *hum  = (int16_t)(((int32_t)hraw * 10000) >> 16);
+            *hum = hum_cached;
         }
     }
 
     return status;
 }
 
+int hdc1000_read_cached(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
+{
+    /* check if initialization phase or outdated */
+    if ((last_read_time == 0xFFFFFFFF && hum_cached == 0 && temp_cached == 0) || 
+        (xtimer_now_usec() - last_read_time > HDC1000_RENEW_INTERVAL)) {
+        /* update last_read_time */
```

Comment:
>This initial values are to check initial condition in hdc1000_read_cached()

1. Why do you need to differentiate between initial and outdated condition?
2. Can you explain why previous proposals doe not suffice? Cause I'd prefer it that way:
```

last_read = 0
hdc1000_get_results_cached(){
    now = xtimer_now_usec();
    if (now > (last_read + HDC1000_RENEW_INTERVAL)) {
        hdc1000_get_results()
        last_read = now
    }
```

Reply:
When timer is nearly saturated, 0xFFFFFFFF, it is possible that last_read + HDC1000_RENEW_INTERVAL = 0xFFFFFFEE (or some value close to the maximum value) while now is 0x00000012 (some value restarting from 0). 

In this case, now > last_read + HDC1000_RENEW_INTERVAL, but we will get the reverse result.

Reply:
Well, both checks on each read operation might not be the optimum from the performance perspective but I'm fine with it. We can still optimize in follow-up PRs.